x.c

```diff
@@ -2814,6 +2877,29 @@ run(void)
 			timeout = (timeout >= 0) ? MIN(timeout, cursortimeout) : cursortimeout;
 		}
 
+		/*
+		 * The distance between the mouse and the window is x.
+		 * The next scroll time is reduced by y.
+		 * y = x^2 - 2*x.
+		 * target_scrolltimeout is the final timeout value which minimum value is 1.
+		*/
+		timeout = asr.iscontinue ? minlatency : timeout;
+		target_scrolltimeout = disable_autoscroll_accelerate ? autoscrolltimeout : MAX(autoscrolltimeout - (asr.speed) * (asr.speed) - (asr.speed) * 2, 1);
+		if (asr.iscontinue == 1 && (asr.dir == SCROLL_UP || asr.dir == SCROLL_DOWN)) {
+			if (target_scrolltimeout - TIMEDIFF(now, lastscroll) <= 0) {
+				if (asr.dir == SCROLL_UP)
+					kscrollup(&(Arg){.i = 1});
+				else
+					kscrolldown(&(Arg){.i = 1});
+				xyselextend(asr.buttonx, asr.buttony, asr.seltype);
+				lastscroll = now;
+			}
+			asr.isscrolling = 1;
+		}
+		else {
+			asr.isscrolling = 0;
+		}
+
```

Comment:
```suggestion
		/*
		 * The distance between the mouse and the window is x.
		 * The next scroll time is reduced by y.
		 * y = x^2 - 2*x.
		 * target_scrolltimeout is the final timeout value which minimum value is 1.
		*/
		if (asr.iscontinue == 1) {
			target_scrolltimeout = disable_autoscroll_accelerate
				? autoscrolltimeout
				: MAX((double)autoscrolltimeout - (asr.speed) * (asr.speed) - (asr.speed) * 2, 1.0);
			if (target_scrolltimeout - TIMEDIFF(now, lastscroll) <= 0) {
				if (asr.dir == SCROLL_UP)
					kscrollup(&(Arg){.i = 1});
				else
					kscrolldown(&(Arg){.i = 1});
				xyselextend(asr.buttonx, asr.buttony, asr.seltype);
				lastscroll = now;
			}
			timeout = target_scrolltimeout - TIMEDIFF(now, lastscroll);
			asr.isscrolling = 1;
		}
		else {
			asr.isscrolling = 0;
		}

```

Notes.
* I simplified the outer if statement because the asr.dir variable only has two states.
* I changed the autoscrolltimeout variable to double.
* Timeouts are calculated inside the if statement.
* We can override the timeout variable because screen scrolling is the most important timer.

Reply:
`timeout = target_scrolltimeout - TIMEDIFF(now, lastscroll);`
The judgment here is not perfect, we must ensure that timeout is `> 0`, otherwise on the next loop, it will be blocked in the above `pselect` and cannot scroll again.